Python/Product/PythonTools/PythonTools/Project/PytestConfigureInfoBar.cs

```diff
@@ -0,0 +1,222 @@
+            )));
+
+            actions.Add(configureActionItem);
+            actions.Add(new InfoBarHyperlink(Strings.ConfigurePytestTextInfoBarIgnoreAction, (Action)Ignore));
+
+            LogEvent(ConfigurePytestInfoBarActions.Prompt);
+            Create(new InfoBarModel(messages, actions, KnownMonikers.StatusInformation));
+        }
+
+        private async void InstallAndEnablePytest() {
+            LogEvent(ConfigurePytestInfoBarActions.EnableAndInstall);
+
+            //IS there a better way to do this? Combining two link queries? TODO
+            Boolean pytestInstalled = false;
+            var packageManagers = _interpreterOptionsService.GetPackageManagers(_interpreterFactory);
+            foreach (var packageManager in packageManagers) {
+                pytestInstalled = await packageManager.InstallAsync(
+                    new PackageSpec("pytest"),
+                    new VsPackageManagerUI(Site),
+                    CancellationToken.None
+                    );
+
+                if (pytestInstalled) {
+                    SetProperty(PythonConstants.TestFrameworkSetting, "Pytest");
+                    break;
+                }
+            }
+
+            if (!pytestInstalled) {
+                //Pytest installation failed.Show error message? TODO
+            }
+
+            Close();
+        }
+
+        private void EnablePytest() {
+            LogEvent(ConfigurePytestInfoBarActions.Enable);
+            SetProperty(PythonConstants.TestFrameworkSetting, "Pytest");
+            Close();
+        }
+
+        private void Ignore() {
+            LogEvent(ConfigurePytestInfoBarActions.Ignore);
+            SetProperty(PythonConstants.SuppressPytestConfigPrompt, true.ToString());
+            Close();
+        }
+
+        private void LogEvent(string action) {
+            Logger?.LogEvent(
+                PythonLogEvent.ConfigurePytestInfoBar,
+                new ConfigurePytestInfoBarInfo() {
+                    Action = action,
+                    Context = InfoBarContext
+                }
+            );
+        }
+
+        protected abstract void SetProperty(string propertyName, string propertyValue);
+
+        protected TestFrameworkType GetTestFramework(string propertyValue) {
+            if (Enum.TryParse(
+                propertyValue,
+                ignoreCase: true,
+                out TestFrameworkType parsedFramework)
+            ) {
+                return parsedFramework;
+            }
+
+            return TestFrameworkType.None;
+        }
+    }
+
+    internal sealed class ConfigurePytestProjectInfoBar : ConfigurePytestInfoBar {
+        private PythonProjectNode Project { get; }
+
+        public ConfigurePytestProjectInfoBar(IServiceProvider site, PythonProjectNode projectNode) : base(site) {
+            Project = projectNode ?? throw new ArgumentNullException(nameof(projectNode));
+        }
+
+        public override async Task CheckAsync() {
+            await CheckAsync(
```

Comment:
this could be just return CheckAsync(..) and you could remove the async in the function definition

https://www.youtube.com/watch?v=J0mcYVxJEl0&t=2586s

Reply:
Can you clarify? Isn't that just fire and forget approach which is discouraged?

CheckAsync is an abstract method in PythonInfoBar which is being overriden, so I can't change the signature.